### ISSUE TITLE
How to finetune using fp16 precision?

### ISSUE THREAD

**Issue by mkrupczak3** [roboflow/rf-detr#117]
I saw in `main.py` that there is an argument called `fp16_eval`:
https://github.com/roboflow/rf-detr/blob/5e2979d55e2aab0da80bd8a8ca8cf09c6f4ab909/rfdetr/main.py#L573

I tried setting this as an argument for the RFDETR object constructor:
```python
#!/usr/bin/env python3                                                                                                                                                                           
from rfdetr import RFDETRBase
model = RFDETRBase(resolution=1120)

history = []

def callback2(data):
    history.append(data)

model.callbacks["on_fit_epoch_end"].append(callback2)

model.train(dataset_dir='/hpcuser/combined_coco', epochs=250, batch_size=4, grad_accum_steps=1, fp16_eval=True, lr=1e-4, output_dir='/hpcuser/rf-detr/combined_bravo_1')
```

However, watching nvidia-smi it shows the GPU memory usage is exactly the same as before including the fp16_eval argument.

Am I using this argument incorrectly? How can we perform finetuning under fp16 precision?

**Comment by isaacrob-roboflow** [comment]
mixed precision is supported by default using AMP, and can be enabled or disabled at model initialization time by setting amp=True or False. that enables or disables it during both training and eval. by default it's True.

the fp16_eval argument tries to force the model to use half precision during eval, but I am not sure it does anything meaningful as the forward call is wrapped in autocast during eval. if it is modifying the model in-place as it appears to be trying to, it may actually be making training results worse as full fp16 can damage normalization layers. we adapted this codebase very quickly from our research base and this appears to be a lingering argument. we need to go through and purge all such arguments.

does that answer your question?

**Comment by mkrupczak3** [comment]
I think so.

@isaacrob-roboflow are you sure amp is True by default? I found this code which would seem to suggest otherwise:
https://github.com/roboflow/rf-detr/blob/5e2979d55e2aab0da80bd8a8ca8cf09c6f4ab909/rfdetr/main.py#L594

https://github.com/roboflow/rf-detr/blob/5e2979d55e2aab0da80bd8a8ca8cf09c6f4ab909/rfdetr/main.py#L781

**Comment by isaacrob-roboflow** [comment]
the relevant line is here https://github.com/roboflow/rf-detr/blob/5e2979d55e2aab0da80bd8a8ca8cf09c6f4ab909/rfdetr/config.py#L19

try turning it off and see if it makes a difference

**Comment by isaacrob-roboflow** [comment]
I am marking this as resolved. feel free to reopen if you feel otherwise @mkrupczak3

**Comment by mkrupczak3** [comment]
Thanks for the prompt replies @isaacrob-roboflow . I think it's safe to keep this issue closed
